modules/ichorcna/1.0/ichorcna.smk

```diff
@@ -178,42 +178,107 @@ rule _ichorcna_input_bam:
     output:
         bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
         bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai", # specific to readCounter
+        crai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.crai"
     run:
         op.absolute_symlink(input.bam, output.bam)
         op.absolute_symlink(input.bai, output.bai)
+        op.absolute_symlink(input.bai, output.crai)
      
-# This function will return a comma-separated list of chromosomes to include in readCounter
-def get_chromosomes(wildcards):
-    chromosomes=[]
-    for i in range(1,23):
-        chromosomes.append(str(i))
-    chromosomes.append("X")
-    chromosomes.append("Y")
-    if "38" in str(wildcards.genome_build):
-        chromosomes = ["chr" + x for x in chromosomes]
-    chromosomes= ",".join(chromosomes)    
-    return chromosomes
 
-rule _ichorcna_read_counter:
+# set-up for CRAM files (readCounter does not work with CRAM)
+# deeptools to get .bw from .bam and .cram
+rule _ichorcna_bamCoverage:
     input:
         bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
         bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai",
+        crai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.crai",
         ichorcna_package = CFG["dirs"]["inputs"] + "ichorcna_dependencies_installed.success",
         symlink_complete = ichorDir + "symlink.done"
     output:
-        CFG["dirs"]["readDepth"] + "{seq_type}--{genome_build}/{binSize}/{sample_id}.bin{binSize}.wig"
+        bw = CFG["dirs"]["readDepth"] + "{seq_type}--{genome_build}/{binSize}/bw/{sample_id}.bin{binSize}.bw"
     params:
-        binSize = CFG["options"]["readcounter"]["binSize"],
-        qual = CFG["options"]["readcounter"]["qual"],
-        chrs = get_chromosomes
-    conda: CFG["conda_envs"]["ichorcna"]
-    threads: CFG["threads"]["readcounter"]
+        binSize = CFG["options"]["deeptools"]["binSize"],
+        qual = CFG["options"]["deeptools"]["qual"],
+        excludeFlag = CFG["options"]["deeptools"]["flagExclude"],
+        opt = CFG["options"]["deeptools"]["opt"],
+        dirOut = CFG["dirs"]["readDepth"] + "{seq_type}--{genome_build}/{binSize}/bw/"
+    conda: CFG["conda_envs"]["deeptools"]
+    threads: CFG["threads"]["deeptools"]
     resources:
-        **CFG["resources"]["readcounter"]
+        **CFG["resources"]["deeptools"]
     log:
-        CFG["logs"]["readDepth"] + "{seq_type}--{genome_build}/{binSize}/{sample_id}.bin{binSize}.log"
+        CFG["logs"]["readDepth"] + "{seq_type}--{genome_build}/{binSize}/bw/{sample_id}.bin{binSize}.log"
     shell:
-        "readCounter {input.bam} -c {params.chrs} -w {params.binSize} -q {params.qual} > {output} 2> {log}"
+        """
+            mkdir -p {params.dirOut}; 
```

Comment:
I thought it did but it kept bugging out while running the program (kept saying the output file can't be written, since the directory was not there yet), so I have it in here just to circumvent this issue